homeassistant/components/hassio/__init__.py

```diff
@@ -392,16 +393,6 @@ def get_core_info(hass):
     return hass.data.get(DATA_CORE_INFO)
 
 
-@callback
-@bind_hass
-def is_hassio(hass: HomeAssistant) -> bool:
```

Comment:
This move breaks all custom integrations that might use it.

Reply:
Right, yea that's why I was hoping not to do that. Maybe we should just mark it as deprecated with a removal timeline? There's a lot of internal integrations that use it so I would imagine there's a good amount of custom ones as well?

Reply:
I think we should have a deprecation period here that logs a warning when accessed.
I'm not sure if there is even any that uses it, but if there are those would hard break as this would throw an import error on load.

Reply:
Yea makes sense. So keep the one in `hassio` with the now unused `hass` parameter, mark it as deprecated directing people to the new helper and log a warning with the grace period.

Reply:
It's still available in this file because we import it at the top and the API is the same if called using `hass.components.hassio.is_hassio`.

Do you think that we should add a fallback param that if set, prints a warning?

```python
def is_hassio(hass = None):
    if hass is not None:
        _LOGGER.warning("hass param deprecated for is_hassio")
    return ...
```

Reply:
oh... yeah we do, I missed that, so it will not be a hard break, it will just mess some parts up when it complains about arguments, which can throw but better than not loading at all :+1: .
I think this is a good option, but we should use the report helper to get the integration frame for the log.

Reply:
I used `@deprecate_function`, is that the one you meant @ludeeus or is there a different on?

Reply:
`get_integration_frame` and `report_integration` from `homeassistant.helpers.frame` is what I meant :+1: